Cabal/src/Distribution/Simple/Setup/Build.hs

```diff
@@ -18,7 +21,15 @@
 -- Definition of the build command-line options.
 -- See: @Distribution.Simple.Setup@
 module Distribution.Simple.Setup.Build
-  ( BuildFlags (..)
+  ( BuildFlags
+      ( BuildCommonFlags
+      , buildVerbosity
+      , buildDistPref
+      , buildCabalFilePath
+      , buildWorkingDir
+      , buildTargets
+      , ..
+      )
```

Comment:
**Breaking change** to Cabal library users. :boom: 

```haskell
import Distribution.Simple.Setup (BuildFlags (buildVerbosity))

miniRepro :: BuildFlags
miniRepro = emptyBuildFlags {buildVerbosity = undefined}
```

— compiles against Cabal 3.12; yields compile error with 3.14:

```
src/Distribution/Extra/Doctest.hs:235:13: error: [GHC-16444]
    • non-bidirectional pattern synonym ‘Cabal-3.14.0.0:Distribution.Simple.Setup.Build.BuildCommonFlags’ used in an expression
    • In a record update at field ‘buildVerbosity’
      and with pattern synonym ‘Cabal-3.14.0.0:Distribution.Simple.Setup.Build.BuildCommonFlags’.
      In the expression: emptyBuildFlags {buildVerbosity = undefined}
      In an equation for ‘miniRepro’:
          miniRepro = emptyBuildFlags {buildVerbosity = undefined}
    |
235 | miniRepro = emptyBuildFlags {buildVerbosity = undefined}
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```